### PR TITLE
Webprofiler add status code to search form

### PR DESCRIPTION
**Pull request by oktapodia**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | https://github.com/symfony/symfony/pull/13034 |
| License | MIT |

With this PR, in the web profiler, you can filter by HTTP status codes.

_Before filter_
image

_After filter_
image

### CODE REVIEW

**Comment by oktapodia** [comment]
Following the PR https://github.com/symfony/symfony/pull/15111

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php [hunk 1]
```diff
@@ -216,19 +216,21 @@ public function searchBarAction(Request $request)
...
             $token = null;
         } else {
-            $ip = $request->query->get('ip', $session->get('_profiler_search_ip'));
-            $method = $request->query->get('method', $session->get('_profiler_search_method'));
-            $url = $request->query->get('url', $session->get('_profiler_search_url'));
-            $start = $request->query->get('start', $session->get('_profiler_search_start'));
-            $end = $request->query->get('end', $session->get('_profiler_search_end'));
-            $limit = $request->query->get('limit', $session->get('_profiler_search_limit'));
-            $token = $request->query->get('token', $session->get('_profiler_search_token'));
+            $ip = $session->get('_profiler_search_ip');
```
Comment (sstok) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: Are these changes correct?

File: src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php [hunk 2]
```diff
@@ -63,12 +63,11 @@ public function find($ip, $url, $limit, $method, $start = null, $end = null)
         $result = array();
         while (count($result) < $limit && $line = $this->readLineFromFile($file)) {
             $values = str_getcsv($line);
-            list($csvToken, $csvIp, $csvMethod, $csvUrl, $csvTime, $csvParent) = $values;
-            $csvStatusCode = isset($values[6]) ? $values[6] : null;
-
+            list($csvToken, $csvIp, $csvMethod, $csvUrl, $csvTime, $csvParent,$csvStatusCode) = $values;
```
Comment (sstok) [inline src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php]: Missing space after comma.

**Comment by sstok** [comment]
Status: Needs work

**Comment by oktapodia** [comment]
@sstok Oops ! Mistake after the rebase, FIxed.

**Comment by oktapodia** [comment]
Status: Needs review

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/search.html.twig [hunk 3]
```diff
@@ -16,6 +16,11 @@
         </div>
 
         <div class="form-group">
+            <label for="status_code">Status</label>
+            <input type="text" name="status_code" id="status_code" value="{{ status_code }}">
```
Comment (Ener-Getick) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/search.html.twig]: Why not using a ``number`` input here?

File: src/Symfony/Component/HttpKernel/Profiler/Profile.php [hunk 4]
```diff
@@ -181,7 +181,7 @@ public function setStatusCode($statusCode)
     }
 
     /**
-     * @return int
+     * @return string
```
Comment (Ener-Getick) [inline src/Symfony/Component/HttpKernel/Profiler/Profile.php]: Why did you change this?

**Comment by oktapodia** [comment]
ping @sstok Can I have some news please ?

File: src/Symfony/Component/HttpKernel/Tests/Profiler/ProfilerTest.php [hunk 5]
```diff
@@ -59,6 +59,13 @@ public function testFindWorksWithInvalidDates()
         $this->assertCount(0, $profiler->find(null, null, null, null, 'some string', ''));
     }
 
+    public function testFindWorksWithStatusCode()
+    {
+        $profiler = new Profiler($this->storage);
+
+        $this->assertCount(0, $profiler->find(null, null, null, null, null, null, 204));
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Tests/Profiler/ProfilerTest.php]: AFAIU, you do not test anything here. Can you add a test where the count is non-zero?

**Comment by javiereguiluz** [comment]
@oktapodia your pull request looks good and the proposed feature is nice! We just need a final push to finish it. Do you think you can add the test asked for by Fabien? Thanks.

**Comment by stof** [comment]
Status: needs work

**Comment by oktapodia** [comment]
I think there are some problems with your tests.

screen

**Comment by Ener-Getick** [comment]
Try maybe to rebase on master.

**Comment by oktapodia** [comment]
@Ener-Getick Thanks but my PR is not mergeable if I rebase on master

**Comment by Ener-Getick** [comment]
This is a new feature so it will be merged in master anyway. Try to rebase it on 3.0 otherwise.

**Comment by oktapodia** [comment]
Same problem :(

**Comment by oktapodia** [comment]
Status: needs review

**Comment by Ener-Getick** [comment]
It's apparently not related to your PR (see https://github.com/symfony/symfony/branches). So don't worry ;)

**Comment by javiereguiluz** [comment]
Failed tests have been fixed. I've played with this feature and it works as expected:





:+1:

Status: reviewed

**Comment by xabbuh** [comment]
:+1:

**Comment by oktapodia** [comment]
Thanks @javiereguiluz

**Comment by fabpot** [comment]
Thank you @oktapodia.

**Comment by rvanlaak** [comment]
:+1:
